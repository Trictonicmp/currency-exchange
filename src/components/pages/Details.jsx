import { useSelector } from 'react-redux';
import DetailsHeader from '../DetailsHeader';
import WeekHistory from '../WeekHistory';
import styles from '../../css/components/details.module.css';
import backIcon from '../../assets/images/next.png';

const Details = () => {
  const country = useSelector((state) => state.selectedCountry);
  const baseCurrency = useSelector((state) => state.baseCurrency);
  return (
    <section>
      <div className={styles.header}>
        <DetailsHeader
          name={country.name}
          flag={country.flag}
          todaysExchange={country.todaysExchange}
          baseCurrency={baseCurrency}
        />
      </div>
      <div className={styles.historyHeader}>
        <div className="pageContainer">
          <button type="button">
            <img src={backIcon} alt="Go Back" title="Go Back" />
          </button>
          <h2>This week&apos;s exchange</h2>
        </div>
      </div>
      <WeekHistory weekHistory={country.weekHistory} />
    </section>
  );
};

export default Details;
