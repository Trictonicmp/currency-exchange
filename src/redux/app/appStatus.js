// Actions
const ERROR = 'currency-exchange/app/ERROR';
const SUCCESS = 'currency-exchange/app/SUCCESS';
const LOADING = 'currency-exchange/app/LOADING';

// Action creators
export const errorStatus = (errorMessage) => ({ type: ERROR, status: `ERROR + ${errorMessage}` });
export const successStatus = () => ({ type: SUCCESS, status: 'SUCCESS' });
export const loadingStatus = () => ({ type: LOADING, status: 'LOADING' });

// Reducer
const reducer = (state = 'LOADING', action) => {
  switch (action.type) {
    case ERROR:
      return action.status;
    case SUCCESS:
      return action.status;
    case LOADING:
      return action.status;
    default:
      return state;
  }
};

export default reducer;
