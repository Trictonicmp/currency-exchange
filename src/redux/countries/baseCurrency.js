import { getCountries } from './countries';

// Actions
const CHANGE_BASE = 'currency-exchange/currencies/CHANGE_BASE';

// Action creators
export const changeBase = (baseCurrency) => async (dispatch) => {
  dispatch(getCountries(baseCurrency));
  dispatch({ type: CHANGE_BASE, payload: baseCurrency });
};

// Reducer
const reducer = (state = 'USD', action) => {
  switch (action.type) {
    case CHANGE_BASE:
      return action.payload;
    default:
      return state;
  }
};

export default reducer;
