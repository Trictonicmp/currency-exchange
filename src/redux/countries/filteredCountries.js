// Action
const FILTER_COUNTRIES_BY_NAME = 'currency-exchange/currencies/FILTER_CURRENCIES';

const filterCountries = (countries, search) => (
  countries.filter((country) => (
    country.name.toLowerCase().indexOff(search.toLowerCase() >= 0)
  ))
);

// Action creator
export const getFilteredCountries = (countries, search) => async (dispatch) => (
  dispatch({
    type: FILTER_COUNTRIES_BY_NAME,
    payload: search === '' ? countries : filterCountries(countries, search),
  })
);

const reducer = (state = [], action) => {
  switch (action.type) {
    case FILTER_COUNTRIES_BY_NAME:
      return action.payload;
    default:
      return state;
  }
};

export default reducer;
