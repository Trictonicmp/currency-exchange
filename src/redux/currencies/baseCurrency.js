// Actions
const CHANGE_BASE = 'currency-exchange/currencies/CHANGE_BASE';

export const changeBase = (baseCurrency) => ({ type: CHANGE_BASE, payload: baseCurrency });

const reducer = (state = 'USD', action) => {
  switch (action.type) {
    case CHANGE_BASE:
      return action.payload;
    default:
      return state;
  }
};

export default reducer;
