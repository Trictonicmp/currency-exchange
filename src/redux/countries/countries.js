import { errorStatus, loadingStatus, successStatus } from '../app/appStatus';
import getCountriesCurrencies from '../../API/CurrencyCountriesAPI';

// Actions
const GET_COUNTRIES = 'currency-exchange/currencies/GET_COUNTRIES';

// Action creators
export const getCountries = (baseCurrency) => async (dispatch) => {
  let countriesCurrencies = [];
  try {
    dispatch(loadingStatus());
    countriesCurrencies = await getCountriesCurrencies(baseCurrency);
  } catch (error) {
    dispatch(errorStatus(error));
  }

  dispatch({
    type: GET_COUNTRIES,
    payload: countriesCurrencies,
  });
  dispatch(successStatus());
};

// Reducer
const initialState = [];
const reducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_COUNTRIES:
      // do something
      return action.payload;
    default:
      return state;
  }
};

export default reducer;
