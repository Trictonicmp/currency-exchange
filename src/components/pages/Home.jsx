import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { getFilteredCountries } from '../../redux/countries/filteredCountries';

const Home = (props) => {
  const { countries } = props;
  const dispatch = useDispatch();
  dispatch(getFilteredCountries(countries, 'Mex'));

  return (
    <section>
      <h1>Home</h1>
      {
        // appStatus === 'LOADING' ? <p>loading</p> : <p>loaded</p>
        'Hey'
      }
    </section>
  );
};

Home.propTypes = {
  countries: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    flag: PropTypes.string,
    todaysExchange: PropTypes.number,
    weekHistory: PropTypes.arrayOf(PropTypes.shape({
      date: PropTypes.string,
      exchange: PropTypes.number,
    })),
  })).isRequired,
};

export default Home;
