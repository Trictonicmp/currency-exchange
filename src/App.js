import { Routes, Route } from 'react-router-dom';
import Details from './components/pages/Details';
import Home from './components/pages/Home';

function App() {
  return (
    <Routes>
      <Route path="/" element={<Home />} />
      <Route path="details" element={<Details />} />
    </Routes>
  );
}

export default App;
