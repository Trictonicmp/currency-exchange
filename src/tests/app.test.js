import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import countries from '../redux/countries/countries';
import appStatus from '../redux/app/appStatus';
import baseCurrency from '../redux/countries/baseCurrency';
import filteredCountries, { getFilteredCountries } from '../redux/countries/filteredCountries';
import selectedCountry from '../redux/selectedItem/selectedITem';
import App from '../App';
import Home from '../components/pages/Home';
import Details from '../components/pages/Details';

const { configureStore } = require('@reduxjs/toolkit');

const createStore = () => {
  const preloadedState = {
    appStatus: 'SUCCESS',
    baseCurrency: 'USD',
    countries: [
      {
        name: 'Afghanistan',
        currencyCode: 'AFN',
        flag: 'ðŸ‡¦ðŸ‡«',
        todaysExchange: 89.230209,
        weekHistory: [
          {
            date: '2022-06-17',
            exchange: 88.9083,
          },
          {
            date: '2022-06-16',
            exchange: 89.272511,
          },
        ],
      },
      {
        name: 'Mexico',
        currencyCode: 'MXN',
        flag: 'ðŸ‡¦ðŸ‡«ðŸ‡²ðŸ‡½',
        todaysExchange: 20.24042,
        weekHistory: [
          {
            date: '2022-06-17',
            exchange: 20.423423,
          },
          {
            date: '2022-06-16',
            exchange: 20.316285,
          },
        ],
      },
    ],

    filteredCountries: [
      {
        name: 'Afghanistan',
        currencyCode: 'AFN',
        flag: 'ðŸ‡¦ðŸ‡«',
        todaysExchange: 89.230209,
        weekHistory: [
          {
            date: '2022-06-17',
            exchange: 88.9083,
          },
          {
            date: '2022-06-16',
            exchange: 89.272511,
          },
        ],
      },
      {
        name: 'Mexico',
        currencyCode: 'MXN',
        flag: 'ðŸ‡¦ðŸ‡«ðŸ‡²ðŸ‡½',
        todaysExchange: 20.24042,
        weekHistory: [
          {
            date: '2022-06-17',
            exchange: 20.423423,
          },
          {
            date: '2022-06-16',
            exchange: 20.316285,
          },
        ],
      },
    ],

    selectedCountry: {
      name: 'Mexico',
      currencyCode: 'MXN',
      flag: 'ðŸ‡¦ðŸ‡«ðŸ‡²ðŸ‡½',
      todaysExchange: 20.24042,
      weekHistory: [
        {
          date: '2022-06-17',
          exchange: 20.423423,
        },
        {
          date: '2022-06-16',
          exchange: 20.316285,
        },
      ],
    },
  };

  return configureStore({
    preloadedState,
    reducer: {
      countries,
      appStatus,
      baseCurrency,
      filteredCountries,
      selectedCountry,
    },
  });
};

describe('App snapshots', () => {
  test('It should render the app', () => {
    const store = createStore();
    const app = render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>,
    );

    expect(app).toMatchSnapshot();
  });

  test('It should render the Home', () => {
    const countries = [
      {
        name: 'Afghanistan',
        currencyCode: 'AFN',
        flag: 'ðŸ‡¦ðŸ‡«',
        todaysExchange: 89.230209,
        weekHistory: [
          {
            date: '2022-06-17',
            exchange: 88.9083,
          },
          {
            date: '2022-06-16',
            exchange: 89.272511,
          },
        ],
      },
      {
        name: 'Mexico',
        currencyCode: 'MXN',
        flag: 'ðŸ‡¦ðŸ‡«ðŸ‡²ðŸ‡½',
        todaysExchange: 20.24042,
        weekHistory: [
          {
            date: '2022-06-17',
            exchange: 20.423423,
          },
          {
            date: '2022-06-16',
            exchange: 20.316285,
          },
        ],
      },
    ];
    const store = createStore();
    const app = render(
      <Provider store={store}>
        <BrowserRouter>
          <Home countries={countries} baseCurrency="USD" />
        </BrowserRouter>
      </Provider>,
    );

    expect(app).toMatchSnapshot();
  });

  test('It should render the Details', () => {
    const store = createStore();
    const app = render(
      <Provider store={store}>
        <BrowserRouter>
          <Details />
        </BrowserRouter>
      </Provider>,
    );

    expect(app).toMatchSnapshot();
  });
});

describe('App events', () => {
  test('Click on country item should open details page', () => {
    const store = createStore();
    render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>,
    );
    fireEvent.click(screen.getByText('Mexico'));
    expect(screen.getByText("This week's exchange")).toBeInTheDocument();
  });
});
