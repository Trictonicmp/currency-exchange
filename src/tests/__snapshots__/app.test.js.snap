// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshots It should render the Details 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <div
          class="header"
        >
          <div
            class="pageContainer detailsHeader"
          >
            <h3>
              ðŸ‡¦ðŸ‡«ðŸ‡²ðŸ‡½
            </h3>
            <div>
              <span>
                Country
              </span>
              <h2>
                Mexico
              </h2>
            </div>
            <div>
              <span>
                Currency
              </span>
              <h2>
                MXN
              </h2>
            </div>
            <div>
              <span>
                Today's exchange | based on: 1 USD
              </span>
              <h2>
                20.24042
              </h2>
            </div>
          </div>
        </div>
        <div
          class="historyHeader"
        >
          <div
            class="pageContainer"
          >
            <button
              type="button"
            >
              <img
                alt="Go Back"
                src="next.png"
                title="Go Back"
              />
            </button>
            <h2>
              This week's exchange
            </h2>
          </div>
        </div>
        <ul
          class="pageContainer pagecontainer"
        >
          <li
            class="weekItem"
          >
            <div
              class="itemDate"
            >
              <span>
                Year - Month - Day
              </span>
              <p>
                2022-06-16
              </p>
            </div>
            <div
              class="itemDate"
            >
              <span>
                Exchange
              </span>
              <p
                class="itemExchange"
              >
                20.316285
              </p>
            </div>
          </li>
          <li
            class="weekItem"
          >
            <div
              class="itemDate"
            >
              <span>
                Year - Month - Day
              </span>
              <p>
                2022-06-17
              </p>
            </div>
            <div
              class="itemDate"
            >
              <span>
                Exchange
              </span>
              <p
                class="itemExchange"
              >
                20.423423
              </p>
            </div>
          </li>
        </ul>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <div
        class="header"
      >
        <div
          class="pageContainer detailsHeader"
        >
          <h3>
            ðŸ‡¦ðŸ‡«ðŸ‡²ðŸ‡½
          </h3>
          <div>
            <span>
              Country
            </span>
            <h2>
              Mexico
            </h2>
          </div>
          <div>
            <span>
              Currency
            </span>
            <h2>
              MXN
            </h2>
          </div>
          <div>
            <span>
              Today's exchange | based on: 1 USD
            </span>
            <h2>
              20.24042
            </h2>
          </div>
        </div>
      </div>
      <div
        class="historyHeader"
      >
        <div
          class="pageContainer"
        >
          <button
            type="button"
          >
            <img
              alt="Go Back"
              src="next.png"
              title="Go Back"
            />
          </button>
          <h2>
            This week's exchange
          </h2>
        </div>
      </div>
      <ul
        class="pageContainer pagecontainer"
      >
        <li
          class="weekItem"
        >
          <div
            class="itemDate"
          >
            <span>
              Year - Month - Day
            </span>
            <p>
              2022-06-16
            </p>
          </div>
          <div
            class="itemDate"
          >
            <span>
              Exchange
            </span>
            <p
              class="itemExchange"
            >
              20.316285
            </p>
          </div>
        </li>
        <li
          class="weekItem"
        >
          <div
            class="itemDate"
          >
            <span>
              Year - Month - Day
            </span>
            <p>
              2022-06-17
            </p>
          </div>
          <div
            class="itemDate"
          >
            <span>
              Exchange
            </span>
            <p
              class="itemExchange"
            >
              20.423423
            </p>
          </div>
        </li>
      </ul>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App snapshots It should render the Home 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <div
          class="filtersSection"
        >
          <div
            class="pageContainer pageContainer"
          >
            <h2>
              Base Currency
            </h2>
            <select
              class="select"
            >
              <option
                value="AFN"
              >
                AFN - Afghanistan
              </option>
              <option
                value="MXN"
              >
                MXN - Mexico
              </option>
            </select>
            <input
              class="search"
              placeholder="Search"
              type="text"
            />
          </div>
        </div>
        <div
          class="countriesListContainer"
        >
          <ul
            class="countriesList"
          >
            <li
              class="listItem"
            >
              <a
                class="
        pageContainer
        pageContainer
        listItemDetails
        "
                href="/details?country=Afghanistan"
              >
                <div
                  class="countryDetails"
                >
                  <span
                    class="countryFlag"
                  >
                    ðŸ‡¦ðŸ‡«
                  </span>
                  <h3
                    class="countryName"
                  >
                    Afghanistan
                  </h3>
                </div>
                <div
                  class="countryExchangeContainer"
                >
                  <p>
                    Exchange
                  </p>
                  <div
                    class="countryExchange"
                  >
                    <span>
                      1 USD
                    </span>
                    =
                    <span>
                      89.230209 AFN
                    </span>
                  </div>
                </div>
                <img
                  alt=">"
                  class="detailsIcon"
                  src="right-arrow.png"
                />
              </a>
            </li>
            <li
              class="listItem"
            >
              <a
                class="
        pageContainer
        pageContainer
        listItemDetails
        "
                href="/details?country=Mexico"
              >
                <div
                  class="countryDetails"
                >
                  <span
                    class="countryFlag"
                  >
                    ðŸ‡¦ðŸ‡«ðŸ‡²ðŸ‡½
                  </span>
                  <h3
                    class="countryName"
                  >
                    Mexico
                  </h3>
                </div>
                <div
                  class="countryExchangeContainer"
                >
                  <p>
                    Exchange
                  </p>
                  <div
                    class="countryExchange"
                  >
                    <span>
                      1 USD
                    </span>
                    =
                    <span>
                      20.24042 MXN
                    </span>
                  </div>
                </div>
                <img
                  alt=">"
                  class="detailsIcon"
                  src="right-arrow.png"
                />
              </a>
            </li>
          </ul>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <div
        class="filtersSection"
      >
        <div
          class="pageContainer pageContainer"
        >
          <h2>
            Base Currency
          </h2>
          <select
            class="select"
          >
            <option
              value="AFN"
            >
              AFN - Afghanistan
            </option>
            <option
              value="MXN"
            >
              MXN - Mexico
            </option>
          </select>
          <input
            class="search"
            placeholder="Search"
            type="text"
          />
        </div>
      </div>
      <div
        class="countriesListContainer"
      >
        <ul
          class="countriesList"
        >
          <li
            class="listItem"
          >
            <a
              class="
        pageContainer
        pageContainer
        listItemDetails
        "
              href="/details?country=Afghanistan"
            >
              <div
                class="countryDetails"
              >
                <span
                  class="countryFlag"
                >
                  ðŸ‡¦ðŸ‡«
                </span>
                <h3
                  class="countryName"
                >
                  Afghanistan
                </h3>
              </div>
              <div
                class="countryExchangeContainer"
              >
                <p>
                  Exchange
                </p>
                <div
                  class="countryExchange"
                >
                  <span>
                    1 USD
                  </span>
                  =
                  <span>
                    89.230209 AFN
                  </span>
                </div>
              </div>
              <img
                alt=">"
                class="detailsIcon"
                src="right-arrow.png"
              />
            </a>
          </li>
          <li
            class="listItem"
          >
            <a
              class="
        pageContainer
        pageContainer
        listItemDetails
        "
              href="/details?country=Mexico"
            >
              <div
                class="countryDetails"
              >
                <span
                  class="countryFlag"
                >
                  ðŸ‡¦ðŸ‡«ðŸ‡²ðŸ‡½
                </span>
                <h3
                  class="countryName"
                >
                  Mexico
                </h3>
              </div>
              <div
                class="countryExchangeContainer"
              >
                <p>
                  Exchange
                </p>
                <div
                  class="countryExchange"
                >
                  <span>
                    1 USD
                  </span>
                  =
                  <span>
                    20.24042 MXN
                  </span>
                </div>
              </div>
              <img
                alt=">"
                class="detailsIcon"
                src="right-arrow.png"
              />
            </a>
          </li>
        </ul>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App snapshots It should render the app 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <div
          class="pageContainer"
        >
          <a
            aria-current="page"
            class="logo active"
            href="/"
            title="Go home"
          >
            CurrencyXchange
          </a>
        </div>
      </header>
      <section>
        <div
          class="filtersSection"
        >
          <div
            class="pageContainer pageContainer"
          >
            <h2>
              Base Currency
            </h2>
            <select
              class="select"
            >
              <option
                value="AFN"
              >
                AFN - Afghanistan
              </option>
              <option
                value="MXN"
              >
                MXN - Mexico
              </option>
            </select>
            <input
              class="search"
              placeholder="Search"
              type="text"
            />
          </div>
        </div>
        <div
          class="countriesListContainer"
        >
          <ul
            class="countriesList"
          >
            <li
              class="listItem"
            >
              <a
                class="
        pageContainer
        pageContainer
        listItemDetails
        "
                href="/details?country=Afghanistan"
              >
                <div
                  class="countryDetails"
                >
                  <span
                    class="countryFlag"
                  >
                    ðŸ‡¦ðŸ‡«
                  </span>
                  <h3
                    class="countryName"
                  >
                    Afghanistan
                  </h3>
                </div>
                <div
                  class="countryExchangeContainer"
                >
                  <p>
                    Exchange
                  </p>
                  <div
                    class="countryExchange"
                  >
                    <span>
                      1 USD
                    </span>
                    =
                    <span>
                      89.230209 AFN
                    </span>
                  </div>
                </div>
                <img
                  alt=">"
                  class="detailsIcon"
                  src="right-arrow.png"
                />
              </a>
            </li>
            <li
              class="listItem"
            >
              <a
                class="
        pageContainer
        pageContainer
        listItemDetails
        "
                href="/details?country=Mexico"
              >
                <div
                  class="countryDetails"
                >
                  <span
                    class="countryFlag"
                  >
                    ðŸ‡¦ðŸ‡«ðŸ‡²ðŸ‡½
                  </span>
                  <h3
                    class="countryName"
                  >
                    Mexico
                  </h3>
                </div>
                <div
                  class="countryExchangeContainer"
                >
                  <p>
                    Exchange
                  </p>
                  <div
                    class="countryExchange"
                  >
                    <span>
                      1 USD
                    </span>
                    =
                    <span>
                      20.24042 MXN
                    </span>
                  </div>
                </div>
                <img
                  alt=">"
                  class="detailsIcon"
                  src="right-arrow.png"
                />
              </a>
            </li>
          </ul>
        </div>
      </section>
      <footer
        class="footer"
      >
        <div
          class="pageContainer"
        >
          <p>
            This Project utilizes both
            <span
              class="badge"
            >
              Rest Countries API
            </span>
            and
            <span
              class="badge"
            >
              Exchangerate.host API
            </span>
            to get countries data and currency exchange data.
            <br />
            <br />
            More info (
            <a
              class="badge linkBadge"
              href="https://restcountries.com/"
            >
              Rest Countries
            </a>
            <a
              class="badge linkBadge"
              href="https://exchangerate.host"
            >
              Exchangerate.host
            </a>
            )
          </p>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <div
        class="pageContainer"
      >
        <a
          aria-current="page"
          class="logo active"
          href="/"
          title="Go home"
        >
          CurrencyXchange
        </a>
      </div>
    </header>
    <section>
      <div
        class="filtersSection"
      >
        <div
          class="pageContainer pageContainer"
        >
          <h2>
            Base Currency
          </h2>
          <select
            class="select"
          >
            <option
              value="AFN"
            >
              AFN - Afghanistan
            </option>
            <option
              value="MXN"
            >
              MXN - Mexico
            </option>
          </select>
          <input
            class="search"
            placeholder="Search"
            type="text"
          />
        </div>
      </div>
      <div
        class="countriesListContainer"
      >
        <ul
          class="countriesList"
        >
          <li
            class="listItem"
          >
            <a
              class="
        pageContainer
        pageContainer
        listItemDetails
        "
              href="/details?country=Afghanistan"
            >
              <div
                class="countryDetails"
              >
                <span
                  class="countryFlag"
                >
                  ðŸ‡¦ðŸ‡«
                </span>
                <h3
                  class="countryName"
                >
                  Afghanistan
                </h3>
              </div>
              <div
                class="countryExchangeContainer"
              >
                <p>
                  Exchange
                </p>
                <div
                  class="countryExchange"
                >
                  <span>
                    1 USD
                  </span>
                  =
                  <span>
                    89.230209 AFN
                  </span>
                </div>
              </div>
              <img
                alt=">"
                class="detailsIcon"
                src="right-arrow.png"
              />
            </a>
          </li>
          <li
            class="listItem"
          >
            <a
              class="
        pageContainer
        pageContainer
        listItemDetails
        "
              href="/details?country=Mexico"
            >
              <div
                class="countryDetails"
              >
                <span
                  class="countryFlag"
                >
                  ðŸ‡¦ðŸ‡«ðŸ‡²ðŸ‡½
                </span>
                <h3
                  class="countryName"
                >
                  Mexico
                </h3>
              </div>
              <div
                class="countryExchangeContainer"
              >
                <p>
                  Exchange
                </p>
                <div
                  class="countryExchange"
                >
                  <span>
                    1 USD
                  </span>
                  =
                  <span>
                    20.24042 MXN
                  </span>
                </div>
              </div>
              <img
                alt=">"
                class="detailsIcon"
                src="right-arrow.png"
              />
            </a>
          </li>
        </ul>
      </div>
    </section>
    <footer
      class="footer"
    >
      <div
        class="pageContainer"
      >
        <p>
          This Project utilizes both
          <span
            class="badge"
          >
            Rest Countries API
          </span>
          and
          <span
            class="badge"
          >
            Exchangerate.host API
          </span>
          to get countries data and currency exchange data.
          <br />
          <br />
          More info (
          <a
            class="badge linkBadge"
            href="https://restcountries.com/"
          >
            Rest Countries
          </a>
          <a
            class="badge linkBadge"
            href="https://exchangerate.host"
          >
            Exchangerate.host
          </a>
          )
        </p>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
